client/src/api/UserAPI.ts

```diff
@@ -52,3 +56,86 @@ export const loginRequest = async (
     throw error;
   }
 };
+
+/**
+ * API Reset Password
+ *
+ * @param {string} username - Username
+ * @param {string} password - Password
+ * @param {string} securityQuestion - Security Question
+ * @param {string} securityAnswer - Security Answer
+ * @returns {AxiosResponse}
+ * @throws {AxiosError}
+ */
+export const resetPassword = async (
+  data: ResetPasswordVals,
+): Promise<AxiosResponse<MessageResponse>> => {
+  try {
+    const response = await axios.post<MessageResponse>(
+      'http://localhost:7005/users/reset-password',
+      data,
+    );
+    return response;
+  } catch (error) {
+    throw error;
+  }
+};
+
+/**
+ * API Update an user's username
+ *
+ * @param {string} username - Username
+ * @returns {AxiosResponse}
+ * @throws {AxiosError}
+ */
+export const updateUsername = async (
+  data: UpdateUsernameVals,
+): Promise<AxiosResponse<MessageResponse>> => {
+  try {
+    const response = await axios.post<MessageResponse>(
+      'http://localhost:7005/users/update-username',
+      data,
+    );
+    return response;
+  } catch (error) {
+    throw error;
+  }
+};
+
+/**
+ * API Update an user's password
+ *
+ * @param {string} password - Password
+ * @returns {AxiosResponse}
+ * @throws {AxiosError}
+ */
+export const updatePassword = async (
+  data: UpdatePasswordVals,
+): Promise<AxiosResponse<MessageResponse>> => {
```

Comment:
I think what we want to show is what does the data looks like, I could change it so that the param is a string and list off what does it have like this?
![image](https://github.com/BarbzCodez/Spendr/assets/93229091/f8c8cef3-a3f1-49c6-a0fd-f443b50818a4)

Reply:
IMO this is clearest and most accurate
```typescript
@param {SignupVals} data - The values entered by the user on sign up
```

Because I think you have to send the param as a `SignupVals` anyway